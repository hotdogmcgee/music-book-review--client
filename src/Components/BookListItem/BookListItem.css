.ratings-container {
    display: flex;
    flex-direction: column;
    font-size: 10px;
    margin-top: 30px
}

.book-list-item > p {
    margin: 0
}

.book-list-item {
    display: flex;
    justify-content: space-between;
    margin-top: 0;
    margin-bottom: 20px
}

.book-list-img-container {
    max-width: 100px
}

.book-info {
    border-left: 1px solid black;
    padding-left: 5px;
    text-align: left;
    width: 70vw
}

.book-info h3 {
    margin-top: 0;
}

.book-info a {
    text-decoration: none;
}



