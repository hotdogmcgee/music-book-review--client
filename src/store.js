export const STORE = {
    bookList: [
        {
            image: 'an image',
            title: 'where the red fern grows',
            author: 'i forgot',
            description: 'two dogs in a very sad tale',
            rating: 4,
            numReviews: 10
        },
        {
            image: 'another image',
            title: 'Harry Potter and the Goblet of Fire',
            author: 'JK Rowling',
            description: 'everyone knows it by now',
            rating: 3,
            numReviews: 3
        }
    ]
}