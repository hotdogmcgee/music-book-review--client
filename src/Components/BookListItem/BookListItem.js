import React from "react";
import './BookListItem.css'


export default function BookListItem(props) {
  const { title, author, rating, description, image, numReviews } = props;

  return (
    <div>
      <p>{image}</p>
      <div className="book-info">
        <h3>{title}</h3>
        <p>{author}</p>
        <p>{description}</p>
      </div>
      <div className="ratings-container">
      <span>{rating}</span>
      <span>Based on {numReviews} reviews</span>
        

      </div>
    </div>
  );
}
