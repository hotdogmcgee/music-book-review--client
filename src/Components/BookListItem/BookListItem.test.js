import React from 'react';
import ReactDOM from 'react-dom';
import BookListItem from './BookListItem.js'
import renderer from 'react-test-renderer'
import { STORE } from '../../store.js'


describe('BookList component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BookListItem {...STORE.bookList[0]}/>, div);
    ReactDOM.unmountComponentAtNode(div); 
    })

    it('renders the UI as expected,', () => {
        const tree = renderer
          .create(<BookListItem {...STORE.bookList[0]}/>)
          .toJSON();
          expect(tree).toMatchSnapshot();
      })
})