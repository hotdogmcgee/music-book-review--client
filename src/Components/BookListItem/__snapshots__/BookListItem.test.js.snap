// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component renders the UI as expected, 1`] = `
<div>
  <p>
    an image
  </p>
  <div
    className="book-info"
  >
    <h3>
      where the red fern grows
    </h3>
    <p>
      i forgot
    </p>
    <p>
      two dogs in a very sad tale
    </p>
  </div>
  <div
    className="ratings-container"
  >
    <span>
      4
    </span>
    <span>
      Based on 
      10
       reviews
    </span>
  </div>
</div>
`;
